### ISSUE TITLE
Improve haxm_check.exe

### ISSUE THREAD

**Issue by soul4soul** [intel/haxm#208]
Update `haxm_check.exe` to include information which indicate that the machine is in a good state to install haxm. The installer provides some of this information. I'd rather not have to run two exe to find out my system isn't in a state for haxm to be used.

**To Reproduce**

1. Run `haxm_check.exe`

See output says
```
VT support -- yes
NX support -- yes
```
However VT may not be enabled in the BIOS so when you go to install haxm it will fail.

**Expected Behavior**

Improved output to include additional information about the environment being ready for install. At a minimum I would like to see `VT enabled`. I'd like to think it shouldn't be hard to do as the installer already performs the check for VT enabled.

```
VT support -- yes
NX support -- yes
VT enabled --yes
```
Consider including hyper-v disabled -- yes/no
Consider including sandbox disabled -- yes/no

It would be nice for one tool to provide all of the information. Would it make sense to get rid of haxm_check.exe and put all of these checks into a page of installer wizard?

**Comment by wcwang** [comment]
Thanks for your report and that is really a good suggestion!

Actually, we have started the development task on this aspect. The new design contains below check points.

CPU vendor: GenuineIntel
VMX supported: Yes
VMX enabled: Yes
EPT supported: Yes
OS version: Windows 10 Pro 10.0.17134
OS architecture: x86_64
Hyper-V disabled: Yes

According to your suggestion, we will consider to implement more check points, such as Sandbox disabled, Guest virtual machines occupied, etc.

After we completed the initial version, we will also consider to open its source code to GitHub as HaxmLoader and welcome the contribution from community.

**Comment by hgkamath** [comment]
also, the haxm version perhaps ?

**Comment by hgkamath** [comment]
true, I just mentioned it as an executable-file with a name '' was a natural place for a user to look. It was my first instinct.

1. <- this I knew, as it became my next resort after the instinctual one.
2. -> this is new to me, I didn't think to look in the bat file, but yes. the batch file queries the registry. One hopes that the haxm in memory / loaded by OS is the same as that in registry key.
3. the "release notes", "intelhaxm-android.exe"/properties/details tab also has the version number, but these are presinstalltion static information.

During an upgrade usually, the following may happen
(a) i stop any qemu-VM (b) update haxm (c) check version if udpated (d) restart vm

This sort of need to double-check version arises, when one want to be doubly sure that haxm got updated soon after an upgrade over a previous version.  The line of thiking is like if the haxm loaded has updated, no need to reboot.

**Comment by wcwang** [comment]
Thanks for your suggestion. We will consider this feature in future release.

**Comment by zarac** [comment]
I concur ; ) (great feature request)

It'd be nice if it also checked if the Windows Feature "Windows Containers" is enabled, since it also conflicts with the installation of HAXM, i think. __I'm not sure about this since i don't know how it works... but i had to disable it in order to get this installation to work.__

Also (why i originally came here to the issue tracker), it'd be nice if the "haxm_silent_run.log" contained information about this. Right now, it (very nicely) informs that Hyper-V conflicts. I "git grepped" through this source to see if i could find the text, update it, and do a pull-request, but i did not find the text. Let me know if you'd like me to create a separate ticket for this.
